subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r55/RunEclipseClosedProjectTasksCrossVersionSpec.groovy

```diff
@@ -0,0 +1,173 @@
+ */
+
+package org.gradle.integtests.tooling.r55
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.integtests.tooling.r54.IntermediateResultHandlerCollector
+import org.gradle.plugins.ide.eclipse.internal.EclipsePluginConstants
+import org.gradle.tooling.model.eclipse.EclipseProject
+import org.gradle.tooling.model.eclipse.EclipseWorkspace
+import org.gradle.tooling.model.eclipse.EclipseWorkspaceProject
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+
+import java.util.regex.Pattern
+
+@TargetGradleVersion(">=5.5")
+@ToolingApiVersion(">=5.5")
+class RunEclipseClosedProjectTasksCrossVersionSpec extends ToolingApiSpecification {
+
+    @Rule
+    TemporaryFolder externalProjectFolder = new TemporaryFolder()
+
+    def "will substitute and run build dependencies for closed projects on startup"() {
+        setup:
+        multiProjectBuildInRootFolder("parent", ["child1", "child2"]) {
+            buildFile << """
+            subprojects {
+                apply plugin: 'java-library'
+            }
+            project(":child1") {
+                configurations {
+                    testArtifacts
+                }
+                task testJar(type: Jar) {
+                    classifier = "tests"
+                }
+                artifacts {
+                    testArtifacts testJar
+                }
+            }
+            project(":child2") {
+                dependencies {
+                    implementation project(":child1");
+                    testImplementation project(path: ":child1", configuration: "testArtifacts")
+                }
+            }
+        """
+        }
+
+        def projectsLoadedHandler = new IntermediateResultHandlerCollector<Void>()
+        def buildFinishedHandler = new IntermediateResultHandlerCollector<EclipseProject>()
+        def out = new ByteArrayOutputStream()
+        def workspace = eclipseWorkspace([gradleProject("child1", false), gradleProject("child2")])
+        when:
+        withConnection { connection ->
+            connection.action().projectsLoaded(new TellGradleToRunBuildDependencyTask(workspace), projectsLoadedHandler)
+                .buildFinished(new LoadEclipseModel(workspace), buildFinishedHandler)
+                .build()
+                .setStandardOutput(out)
+                .forTasks()
+                .run()
+        }
+
+        then:
+        def child2 = buildFinishedHandler.result.children.find { it.name == "child2" }
+        child2.projectDependencies.isEmpty()
+        child2.classpath.collect { it.file.name }.sort() == ['child1-1.0-tests.jar', 'child1-1.0.jar']
+        child2.classpath.find { it.file.name == 'child1-1.0.jar' }.classpathAttributes.find { it.name == EclipsePluginConstants.GRADLE_USED_BY_SCOPE_ATTRIBUTE_NAME }.value == "main,test"
```

Comment:
This seems unrelated to your PR

Reply:
This verifies the correct sourceSet usage calculation in EclipseModelBuilder